src/ngx_lua_resty_lmdb_module.c

```diff
@@ -226,56 +239,44 @@ ngx_lua_resty_lmdb_init_conf(ngx_cycle_t *cycle, void *conf)
 }
 
 
-static ngx_int_t ngx_lua_resty_lmdb_init(ngx_cycle_t *cycle) {
-    /* ngx_lua_resty_lmdb_conf_t *lcf;
-
-    lcf = (ngx_lua_resty_lmdb_conf_t *) ngx_get_conf(cycle->conf_ctx,
-                                                     ngx_lua_resty_lmdb_module); */
-
-    return NGX_OK;
-}
-
-
-static ngx_int_t ngx_lua_resty_lmdb_init_worker(ngx_cycle_t *cycle)
+static ngx_int_t
+ngx_lua_resty_lmdb_create_env(ngx_cycle_t *cycle,
+                              ngx_lua_resty_lmdb_conf_t *lcf,
+                              ngx_flag_t is_master)
 {
-    ngx_lua_resty_lmdb_conf_t *lcf;
     int                        rc;
-    int                        dead;
     MDB_val                    enckey;
     int                        block_size = 0;
     u_char                     keybuf[2048];
 
-    lcf = (ngx_lua_resty_lmdb_conf_t *) ngx_get_conf(cycle->conf_ctx,
-                                                     ngx_lua_resty_lmdb_module);
-
-    if (lcf == NULL || lcf->env_path == NULL) {
-        return NGX_OK;
-    }
 
     rc = mdb_env_create(&lcf->env);
     if (rc != 0) {
         ngx_log_error(NGX_LOG_CRIT, cycle->log, 0,
-                      "unable to create LMDB environment");
+                      "unable to create LMDB environment: %s",
+                      mdb_strerror(rc));
         return NGX_ERROR;
     }
 
     rc = mdb_env_set_mapsize(lcf->env, lcf->map_size);
     if (rc != 0) {
         ngx_log_error(NGX_LOG_CRIT, cycle->log, 0,
-                      "unable to set map size for LMDB");
+                      "unable to set map size for LMDB: %s",
+                      mdb_strerror(rc));
         return NGX_ERROR;
```

Comment:
Do we need a `mdb_close_env` before return `NGX_ERROR`? We can use `goto failed` to deal with others.

Reply:
added.